gooddata-sdk/gooddata_sdk/catalog_model/entity.py

```diff
@@ -0,0 +1,45 @@
+# (C) 2022 GoodData Corporation
+from __future__ import annotations
+
+from typing import Any, Optional
+
+from gooddata_sdk.compute_model import ObjId
+
+
+class CatalogEntity:
+    def __init__(self, entity: dict[str, Any]) -> None:
+        self._e = entity["attributes"]
+        self._entity = entity
+        self._obj_id = ObjId(self._entity["id"], type=self._entity["type"])
+
+    @property
+    def id(self) -> str:
+        return self._entity["id"]
+
+    @property
+    def type(self) -> str:
+        return self._entity["type"]
+
+    @property
+    def title(self) -> Optional[str]:
+        # Optional, not all metadata objects contain title
+        return self._e.get("title")
+
+    @property
+    def description(self) -> Optional[str]:
+        # Optional, not all metadata objects contain description
+        return self._e.get("description")
+
+    @property
+    def obj_id(self) -> ObjId:
+        return self._obj_id
+
+    @classmethod
+    def class_name(cls) -> str:
+        return cls.__name__
```

Comment:
The method is not necessary. It can be replaced by either `cls.__name__` or by `instance.__class__.__name__`